packages/webapp/src/components/Animals/DetailCards/General.tsx

```diff
@@ -70,15 +89,69 @@ const GeneralDetails = ({
     watch,
     getValues,
     resetField,
-    formState: { errors },
+    setValue,
+    formState: { errors, defaultValues },
   } = useFormContext();
 
+  useEffect(() => {
+    if (mode === 'add') {
+      return;
+    }
+    if ((typeOptions && defaultValues?.custom_type_id) || defaultValues?.default_type_id) {
+      const typeId = defaultValues?.custom_type_id
+        ? generateUniqueAnimalId(ANIMAL_ID_PREFIX.CUSTOM, defaultValues?.custom_type_id)
+        : generateUniqueAnimalId(ANIMAL_ID_PREFIX.DEFAULT, defaultValues?.default_type_id);
+      setValue(
+        `${namePrefix}${DetailsFields.TYPE}`,
+        typeOptions.find(({ value }) => value === typeId),
+        { shouldValidate: true },
+      );
+    }
+
+    if ((breedOptions && defaultValues?.custom_breed_id) || defaultValues?.default_breed_id) {
+      const breedId = defaultValues?.custom_breed_id
+        ? generateUniqueAnimalId(ANIMAL_ID_PREFIX.CUSTOM, defaultValues?.custom_breed_id)
+        : generateUniqueAnimalId(ANIMAL_ID_PREFIX.DEFAULT, defaultValues?.default_breed_id);
+
+      setValue(
+        `${namePrefix}${DetailsFields.BREED}`,
+        breedOptions.find(({ value }) => value === breedId),
+      );
+    }
```

Comment:
`transformResponse` seems really handy! I believe it will make the code much cleaner. 
I also love the idea of creating hooks to fetch both default and custom entities!